lib/config/migrations/base/abstract-migration.ts

```diff
@@ -0,0 +1,26 @@
+import type { RenovateConfig } from '../../types';
+import type { Migration } from '../types';
+
+export abstract class AbstractMigration implements Migration {
+  readonly propertyName: string;
+
+  protected readonly originalConfig: RenovateConfig;
+
+  protected readonly migratedConfig: RenovateConfig;
+
+  constructor(
+    propertyName: string,
+    originalConfig: RenovateConfig,
+    migratedConfig: RenovateConfig
+  ) {
+    this.propertyName = propertyName;
+    this.originalConfig = originalConfig;
+    this.migratedConfig = migratedConfig;
+  }
+
+  abstract run(): void;
+
+  protected delete(property: string): void {
```

Comment:
What about making property optional and use `this.PropertyName` then?

This would make some migrations shorter and simpler. 🤗

Reply:
I thought about it as well, but this code `this.delete()` inside migration does not look descriptive for me